java/maven/src/org/netbeans/modules/maven/api/output/OutputUtils.java

```diff
@@ -123,7 +123,7 @@ public static OutputListener matchStackTraceLine(String line, FileObject fileObj
      */
     private static StacktraceAttributes matchStackTraceLine(String line) {
         Matcher match = linePattern.matcher(line);
-        if (match.matches() && match.groupCount() == 3) {
+        if (match.find() && match.groupCount() == 3) {
```

Comment:
NB has a mode where it keeps the log level in the output. It is a checkbox in the maven settings. When that happens this wouldn't match anything, unless this regex would be made aware of the `[level]` prefix.

This regex is public however, I suppose I could change it but we would have to be careful in what way.

Reply:
Not sure I follow. What concretely would the prefix be? Is this something other than the `[INFO] ` etc. matched by `CommandLineOutputHandler.linePattern`?

Reply:
Or maybe more concretely still, in this context, what would examples be of `line` and `linePattern` that would pass `find` but not `matches`?

Reply:
```diff
diff --git a/java/maven/src/org/netbeans/modules/maven/api/output/OutputUtils.java b/java/maven/src/org/netbeans/modules/maven/api/output/OutputUtils.java
index 080fa9c..c8e4378 100644
--- a/java/maven/src/org/netbeans/modules/maven/api/output/OutputUtils.java
+++ b/java/maven/src/org/netbeans/modules/maven/api/output/OutputUtils.java
@@ -50,7 +50,7 @@
  * @author mkleint
  */
 public final class OutputUtils {
-    public static final Pattern linePattern = Pattern.compile("(?:\\[catch\\])?\\sat (.*)\\((?:Native Method|(.*)\\.java\\:(\\d+))\\)"); //NOI18N
+    public static final Pattern linePattern = Pattern.compile("(?:\\[\\w+\\]\\s+)?(?:\\[catch\\])?\\sat (.*)\\((?:Native Method|(.*)\\.java\\:(\\d+))\\)"); //NOI18N
  
     private static final Map<Project, StacktraceOutputListener> projectStacktraceListeners = new WeakHashMap<>();
     private static final Map<FileObject, StacktraceOutputListener> fileStacktraceListeners = new WeakHashMap<>();
@@ -123,7 +123,7 @@
      */
     private static StacktraceAttributes matchStackTraceLine(String line) {
         Matcher match = linePattern.matcher(line);
-        if (match.find() && match.groupCount() == 3) {
+        if (match.matches() && match.groupCount() == 3) {
             String method = match.group(1);
             String file = match.group(2);
             String lineNum = match.group(3);

```
this would work.

Reply:
I see, this is an unrelated constant with the same name. https://github.com/apache/netbeans/blob/2968ee08a5347b727813a8ee4b4215043344c289/java/maven/src/org/netbeans/modules/maven/api/output/OutputUtils.java#L53

Reply:
a line could be
```
[WARN]  	at java.base/java.util.ImmutableCollections.uoe(ImmutableCollections.java:142)
```
for example. Since match goes over the whole line it would need to be made aware of the prefix. I do prefer match too since it has the potential to be faster, esp if an application is printing long lines.

Reply:
> I see, this is an unrelated constant with the same name. 

yes. I actually renamed that already since I got confused too. But its public API - I don't want to cause more trouble as needed.